syntax = "proto3";

package dylk.incidence;

// -------------------------------------------------------------------------
// The Incidences service definition.
service Incidences {
  rpc AlterIncidence   (Incidence)            returns (IncidenceGeneralResponse)  {}
  rpc ListIncidences   (IncidenceParams)      returns (IncidenceListResponse)     {}
  rpc GetIncidence     (IncidenceId)          returns (IncidenceResponse)         {}
}

message Item {
  string equipmentCode = 1;
  int32 quantity       = 2;
}

message Incidence {
  int32  id                  = 1;
  int32  voucherId           = 2;
  string platform            = 3;
  string carrierCode         = 4;
  string patioCode           = 5;
  string observations        = 6;
  string unitCode            = 7;
  string inspectedBy         = 8;
  string operator            = 9;
  string status              = 10;
  repeated Item itemList     = 11;
  double generationTime      = 12;
  double lastTouchTime       = 13;
}

message IncidenceId {
  int32 id = 1;
}

message IncidenceParam {
  string name  = 1;
  string value = 2;
}

message IncidenceParams {
  repeated IncidenceParam paramList     = 1;
  repeated IncidenceParam pageParamList = 2;
}

message IncidenceResponse {
  int32 returnCode       = 1;
  string returnMessage   = 2;
  Incidence incidence    = 3;
}

message IncidenceListResponse {
  int32 returnCode                 = 1;
  string returnMessage             = 2;
  repeated Incidence incidenceList = 3;
  int32 totalItems                 = 4;
  int32 totalPages                 = 5;
}

message IncidenceGeneralResponse {
  int32 returnCode     = 1;
  string returnMessage = 2;
}
